src/main/java/io/anserini/rerank/lib/RocchioReranker.java

```diff
@@ -0,0 +1,211 @@
+      if (context.getSearchArgs().arbitraryScoreTieBreak) {
+        rs = searcher.search(finalQuery, context.getSearchArgs().hits);
+      } else if (context.getSearchArgs().searchtweets) {
+        rs = searcher.search(finalQuery, context.getSearchArgs().hits, BREAK_SCORE_TIES_BY_TWEETID, true);
+      } else {
+        rs = searcher.search(finalQuery, context.getSearchArgs().hits, BREAK_SCORE_TIES_BY_DOCID, true);
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+      return docs;
+    }
+
+    return ScoredDocuments.fromTopDocs(rs, searcher);
+  }
+
+  private FeatureVector estimateRelevanceModel(ScoredDocuments docs, IndexReader reader, boolean tweetsearch, boolean useRf) {
+    FeatureVector f = new FeatureVector();
+
+    int numdocs;
+    if (useRf) {
+      numdocs = docs.documents.length;
+    }
+    else {
+      numdocs = docs.documents.length < fbDocs ? docs.documents.length : fbDocs;
+    }
+
+    List<FeatureVector> docvectors = new ArrayList<>();
+    for (int i = 0; i < numdocs; i++) {
+      if (useRf && docs.scores[i] <= .0) {
+        continue;
+      }
+      try {
+        FeatureVector docVector = createdFeatureVector(
+            reader.getTermVector(docs.ids[i], field), reader, tweetsearch, fbDocs);
+        docVector.pruneToSize(fbTerms);
+        docvectors.add(docVector);
+      } catch (IOException e) {
+        e.printStackTrace();
+        // Just return empty feature vector.
+        return f;
+      }
+    }
+
+    // Precompute the norms once and cache results.
+    float[] norms = new float[docvectors.size()];
+    for (int i = 0; i < docvectors.size(); i++) {
+      norms[i] = (float) docvectors.get(i).computeL1Norm();
+    }
+
+    for (int i = 0; i < docvectors.size(); i++) {
+      float fbWeight = 0.0f;
+      for (String term : docvectors.get(i).getFeatures()) {
+        // Avoids zero-length feedback documents, which causes division by zero when computing term weights.
+        // Zero-length feedback documents occur (e.g., with CAR17) when a document has only terms 
+        // that accents (which are indexed, but not selected for feedback).
+        if (norms[i] > 0.001f) {
+          fbWeight = (docvectors.get(i).getFeatureWeight(term) / norms[i] );
+        }
+        f.addFeatureWeight(term, fbWeight);
+      }
+    }
+
+    f.pruneToSize(fbTerms);
+    f.scaleToUnitL1Norm();
+
+    return f;
+  }
+
+  private FeatureVector createdFeatureVector(Terms terms, IndexReader reader, boolean tweetsearch, int fbDocs) {
+    FeatureVector f = new FeatureVector();
+
+    try {
+      TermsEnum termsEnum = terms.iterator();
+
+      BytesRef text;
+      while ((text = termsEnum.next()) != null) {
+        String term = text.utf8ToString();
+
+        if (term.length() < 2 || term.length() > 20) continue;
+        if (this.filterTerms && !term.matches("[a-z0-9]+")) continue;
+
+        int freq = (int) termsEnum.totalTermFreq();
+        f.addFeatureWeight(term, (float) freq / fbDocs);
```

Comment:
not sure why we're doing this here?

Reply:
Where are we actually getting the BM25 weights of the vector?

Reply:
Updated the freq into BM25 weight